builtin/providers/aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -134,6 +134,12 @@ func resourceAwsLambdaEventSourceMappingRead(d *schema.ResourceData, meta interf
 
 	eventSourceMappingConfiguration, err := conn.GetEventSourceMapping(params)
 	if err != nil {
+		if ec2err, ok := err.(awserr.Error); ok && ec2err.Code() == "ResourceNotFoundException" {
```

Comment:
We are a bit inconsistent about whether we test this directly or write a utility function like `fun isResourceNotFound(err error) bool` - I guess we should decide what the pattern is there and apply that going forward.